src/sentry/integrations/issues.py

```diff
@@ -220,17 +222,29 @@ def create_default_repo_choice(self, default_repo):
         """
         return (default_repo, default_repo)
 
-    def get_annotations(self, group):
-        external_issue_ids = GroupLink.objects.filter(
-            group_id=group.id,
-            project_id=group.project_id,
+    def get_annotations_for_group_list(self, group_list):
+        group_links = GroupLink.objects.filter(
+            group_id__in=[group.id for group in group_list],
+            project_id__in=list(set(group.project.id for group in group_list)),
             linked_type=GroupLink.LinkedType.issue,
             relationship=GroupLink.Relationship.references,
-        ).values_list("linked_id", flat=True)
+        )
 
         external_issues = ExternalIssue.objects.filter(
-            id__in=external_issue_ids, integration_id=self.model.id
+            id__in=[group_link.linked_id for group_link in group_links],
+            integration_id=self.model.id,
         )
+
+        # group annotations by group id
+        annotations_by_group_id = defaultdict(list)
+        for group_link in group_links:
+            issues_for_group = filter(lambda x: x.id == group_link.linked_id, external_issues)
```

Comment:
You could make this a little more efficient if you convert `external_issues` to a `defaultdict(list)` keyed by the linked_id, but not a huge deal.

Reply:
@wedamija The number of `external_issues` should be relatively low so I didn't think it was worth it